app.rb

```diff
@@ -2,31 +2,19 @@
 require 'token_failure_app'
 
 module Api
-  class Base < ::Sinatra::Base
-    register ::Sinatra::Namespace
-    use Rack::PostBodyContentTypeParser
-    include Swagger::Blocks
-
-    helpers ::Api::Helpers::Errors
-    helpers ::Api::Helpers::Auth
-    helpers ::Api::Helpers::Json
-
-    set :app_file, __FILE__
-
-    configure do
-      enable :raise_errors
-      disable :dump_errors, :show_exceptions
-      use Rack::CommonLogger, $logger
-
-      before { content_type(:json) }
-
-      use Warden::Manager do |manager|
-        manager.default_strategies :access_token
-        manager.failure_app = ::TokenFailureApp # lib/token_failure_app.rb
-      end
+  class Application < ::Sinatra::Base
+    # JSON 404's
+    error Sinatra::NotFound do
+      content_type :json
+      halt 404, JSON.dump({
+        error_code: "not_found",
+        message: "Endpoint '#{request.path_info}' not found"
+      })
     end
+
+    use Api::Routes::Main
```

Comment:
I actually don't mind it - gives you a quick break down of whats all in the app... like a very simplified `routes.rb` from rails